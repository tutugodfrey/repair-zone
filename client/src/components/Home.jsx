import React, { Component } from 'react';
import Div from './../components/elementComponents/Div.jsx';
import Link from './../components/elementComponents/Link.jsx';
import Paragraph from './../components/elementComponents/Paragraph.jsx';
import InlineLogin from './InlineLogin.jsx';
import SigninPage from './SigninPage.jsx';
import SignupPage from './SignupPage.jsx';
import store from '../../redux/store';
import actions from '../../redux/actions';

export default class Home extends Component {
  handleSigninLink() {
    store.dispatch(actions.displayPage(SigninPage));
  }

  handleSignupLink() {
    store.dispatch(actions.displayPage(SignupPage));
  }

  homeNav() {
    return <div id='home-nav' className=''>
        <div id='logo-div' className='d-inline'>
          <a>
            <Paragraph paragraphClass='' paragraphId='' content='Repair-Zone' />
          </a>
        </div>
        <div id='home-signin-signup' className='d-inline'>
          <Link linkClass='mr-1' href='#' linkId='#' linkText='Signup' onClick={this.handleSignupLink.bind(this)} />
          <Link linkClass='mr-1' href='#' linkId='#' linkText='Signin' onClick={this.handleSigninLink.bind(this)}/>
        </div>
      </div>
  }

  homeDescription() {
    return <div className='d-flex flex-column'>
      {this.homeNav()}
      <div id='app-description'>
      <h1> Welcome to repair zone</h1><br /><p>We connect Service providers to potential clients</p>
      </div>
      <InlineLogin formClass='form-inline' formId='home-login' />
      </div>
  }

  makeRequest(){
    return <div className='section d-flex flex-column' >
      <h1 className='section-title justify-self-center'>Make Request</h1>
      <p>
        Search for service providers specialize in areas specific to your repair need.
        Make repair or maintenance request specifying your issues
      </p>
    </div>
  }

  trackProgress() {
    return <div className='section' >
      <h1 className='section-title'>Track Progress</h1>
    </div>
  }

  contacts() {
    return <div className='section' >
      <h1 className='section-title'>Messaging</h1>
    </div>
  }

  footer() {
    return <Link
    linkClass='px-5 text-center text-white'
    href = '#'
    linkId='home-signup-btn-2'
    linkText='signup'
    onClick={this.handleSignupLink.bind(this)} />
  }

  render() {
    return (
      <div id='home-page' className='container-fluid'>
        <div className=''>
          <div className=''>
            <Div divId='home-description' divClass='' content={this.homeDescription()}/>
            <Div divId='make-requests' divClass='' content={this.makeRequest()}/>
            <Div divId='track-progress' divClass='' content={this.trackProgress()} />
            <Div divId='contacts' divClass='' content={this.contacts()} />
            <Div divId='home-footer' divClass='' content={this.footer()} />
          </div>
        </div>
      </div>
    )
  }
}
