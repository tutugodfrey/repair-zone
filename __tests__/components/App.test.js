import React from 'react';
import { shallow, mount, render } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';
import ConnectedApp, { App, mapStateToProps } from '../../client/src/components/App.jsx';
import Home from '../../client/src/components/Home';
import { Provider } from 'react-redux';
const mockStore = configureStore();

const initialState = {};
const store = mockStore(initialState);

describe('<App />', () => {
  describe('render()', () => {
    test('renders the component', () => {
      const home = '<div>string of html character</div>'
      const wrapper = shallow(<App store={store} home={home} />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    test.skip('renders the component', () => {
      const localStorage = {
        getItem: jest.fn(),
      }
      store.getState().home = Home;
      const wrapper = mount(
        <Provider store={store}>
          <ConnectedApp
          />
        </Provider>
      );
    });
  });

  test('mapStateToProps should map the component state correctly', () => {
    const appState = {
      home: '<p>some html character</p>',
    }
    const componentState = mapStateToProps(appState);
    expect(componentState.home).toEqual(appState.home)
  })
});