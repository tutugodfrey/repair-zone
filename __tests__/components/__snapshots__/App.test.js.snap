// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> render() renders the component 1`] = `
<div>
  <<div>string of html character</div> />
</div>
`;
